Add windows documentation (#590)
codemagic-ci-cd/codemagic-docs: content/yaml-quick-start/building-a-flutter-app.md

        - other # <-- (Includes APP_STORE_ID)
        # Add the above group environment variables in Codemagic UI (either in Application/Team variables) - https://docs.codemagic.io/variables/environment-variable-groups/
      vars:
        XCODE_WORKSPACE: 'Runner.xcworkspace'
        XCODE_SCHEME: 'Runner'

>         BUNDLE_ID: 'YOUR_BUNDLE_ID_HERE'

Reviewer: just wondering about these changes, could there be a difference between using double and single quotation marks?

          - user_1@example.com
          - user_2@example.com
        notify:
          success: true # To receive a notification when a build succeeds
          failure: false # To not receive a notification when a build fails
      slack:
        # See the following link about how to connect your Slack account - https://docs.codemagic.io/publishing-yaml/distribution/#slack

>         channel: '#builds'

Reviewer: here too, double vs single quotation marks

    7z a -r ../web.zip ./*
```

## Windows desktop builds

### Building an unpackaged Windows executable

```yaml
scripts:
  - name: Build windows
    script: |
      flutter config --enable-windows-desktop
      flutter build windows --release
      cd build/web
      7z a -r ../release.zip ./*
artifacts:
  - build/windows/runner/*.zip
```

### Creating an MSIX package for publishing to Microsoft Store

> Codemagic recommends using the [Flutter msix package](https://pub.dev/packages/msix) for packaging the application. In order for the package to be suitable for publishing to Microsoft Store it is necessary to define certain arguments during packaging.

Reviewer: ```suggestion
Codemagic uses the [Flutter msix package](https://pub.dev/packages/msix) for packaging the application. For publishing to the Microsoft Store, it is necessary to define certain arguments during packaging.
```

> To pass these arguments to the packaging tool, it is necessary to either add the parameters to the packaging command in `codemagic.yaml` or to add the package to your project prior to using Codemagic and [configure](https://pub.dev/packages/msix#gear-configuration-optional) the arguments inside the `pubspec.yaml` file.

Reviewer: ```suggestion
To pass these arguments to the packaging tool, either add the parameters to the packaging command in `codemagic.yaml` or add the package to your project and [configure](https://pub.dev/packages/msix#gear-configuration-optional) the arguments inside the `pubspec.yaml` file.
```

The values for `--display-name`, `--publisher-display-name` and `--publisher` can be found when when logging into [Microsoft Partner Center](https://partner.microsoft.com/en-us/dashboard/home) and navigating to **Apps and games > Your application > Product Identity**.

The argument `--display-name` should be set to match the value of `Package/Identity/Name`, the argument `--publisher` should be set to match the value of `Package/Identity/Publisher` and the argument `--publisher-display-name` should be set to match the value of `Package/Properties/PublisherDisplayName`.

```yaml
scripts:
  - name: Build windows
    script: |
      ...
  - name: Package windows
    script: |
      flutter pub add msix  # this command is reduntant if you have got the msix pub package added to your project already
      flutter pub run msix:create --store --publisher-display-name=MyName --display-name=MyAppName --publisher=CN=xx-yy-zz --identity-name=com.flutter.MyApp --version=1.0.0.1
artifacts:
  - build/windows/**/*.msix
```

> For all the possible flags for the `msix:create` command, check the [pub documentation](https://pub.dev/packages/msix#clipboard-available-configuration-fields), as mentioned prior, these parameters can also be configured in the `pubspec.yaml` file, if the package has been added to the project prior. Take note that when configuring them both inside `codemagic.yaml` and `pubspec.yaml`, the ones inside codemagic.yaml will be prioritized.

Reviewer: ```suggestion
For all the possible flags for the `msix:create` command, check the [pub documentation](https://pub.dev/packages/msix#clipboard-available-configuration-fields). Note that when configuring the flags both in `codemagic.yaml` and `pubspec.yaml`, the ones configured in `codemagic.yaml` take precedence.
```

    7z a -r ../web.zip ./*
```

## Windows desktop builds

{{<notebox>}}
DISCLAIMER: Building Windows applications with Codemagic is in beta stages and will be free to all users with billing enabled until the end of 2021. Any feedback is well received in our [Slack channel](https://slack.codemagic.io).

> {{</notebox>}}

Reviewer: please remove the note
